src/main/scala/sigmastate/serialization/TypeSerializer.scala

```diff
@@ -98,6 +100,12 @@ object TypeSerializer extends ByteBufferSerializer[SType] {
         // `Tuple` type with more than 4 items `(Int, Byte, Box, Boolean, Int)`
         serializeTuple(tup, w)
     }
+    case typeIdent: STypeIdent => {
+      w.put(typeIdent.typeCode)
+      val bytes = typeIdent.name.getBytes(StandardCharsets.UTF_8)
```

Comment:
@greenhat Oh, no, we  need to replace type identifiers with integers. See ValDefs for inspiration.

Reply:
@ergomorphic I knew that something was off with this type. :)